src/net.c

```diff
@@ -99,6 +100,11 @@ static bool mg_aton6(struct mg_str str, struct mg_addr *addr) {
       }
       if (n > 14) return false;
       addr->ip[n] = addr->ip[n + 1] = 0;  // For trailing ::
+    } else if (str.ptr[i] == '%') {       // Scope ID
+      for (i = i + 1; i < str.len; i++) {
+        if (str.ptr[i] < '0' || str.ptr[i] > '9') return false;
+        addr->scope_id *= 10, addr->scope_id += (uint8_t) (str.ptr[i] - '0');
+      }
```

Comment:
The conditional is for the socket code. Not all socket implementations have `scope_id` field in the sockaddr structure.

Reply:
But maybe you're right... Shall we just refuse to build and IPv6 enabled app without scope_id ?
mac, linux and recent windows all have scope_id.
Who does not ? Even lwip has it.

Reply:
I mean: if we parse it as OK and then won't use it, that can be confusing. If the results are not going to be used because scope id has been disabled, I think we should return an error at the parser.